File: repositories/clinic_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type ClinicRepository interface {
	Create(ctx context.Context, clinic *entities.DoctorClinic) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorClinic, error)
	Delete(ctx context.Context, clinicID, doctorID uuid.UUID) error
}

type GormClinicRepository struct {
	db *gorm.DB
}

func NewClinicRepository(db *gorm.DB) ClinicRepository {
	return &GormClinicRepository{db: db}
}

func (r *GormClinicRepository) Create(ctx context.Context, clinic *entities.DoctorClinic) error {
	return r.db.WithContext(ctx).Create(clinic).Error
}

func (r *GormClinicRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorClinic, error) {
	var clinics []entities.DoctorClinic
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Find(&clinics).Error
	return clinics, err
}

func (r *GormClinicRepository) Delete(ctx context.Context, clinicID, doctorID uuid.UUID) error {
	return r.db.WithContext(ctx).Where("clinic_id = ? AND doctor_id = ?", clinicID, doctorID).Delete(&entities.DoctorClinic{}).Error
}

File: python_backend/repositories/clinic_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.clinic import DoctorClinic

class ClinicRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, clinic: DoctorClinic) -> None:
        self.db.add(clinic)
        self.db.commit()
        self.db.refresh(clinic)

    def FindByDoctorID(self, doctor_id: UUID) -> List[DoctorClinic]:
        return self.db.query(DoctorClinic).filter(DoctorClinic.doctor_id == str(doctor_id)).all()

    def Delete(self, clinic_id: UUID, doctor_id: UUID) -> None:
        self.db.query(DoctorClinic).filter(
            DoctorClinic.clinic_id == str(clinic_id),
            DoctorClinic.doctor_id == str(doctor_id),
        ).delete()
        self.db.commit()
